doc/api/http.md

```diff
@@ -630,27 +630,25 @@ deprecated: v13.0.0
 
 See [`request.socket`][].
 
-### `request.end([data[, encoding]][, callback])`
+### `request.end([callback])`
```

Comment:
I think this is a bit problematic because if I remember correctly using only `request.end(chunk)` without any `request.write(chunk)` is a common way to disable the chunked encoding. If this is discouraged the user needs to specify the `Content-Length` header to disable chunked encoding.

Reply:
I think that is fixable, as long as end is called in the same tick, i.e. defer the first write.

Reply:
This could have some intended side effects in http/2, http/3 and and quic... specifically, it would likely end sending additional empty `DATA`/`STREAM` frames to end streams. We'll have to rework some of the internals to account for this change in pattern. It's not too difficult but it's something to keep in mind. It's good that this is a doc-only deprecation for now.

Reply:
/cc @clshortfuse FYI, since you were looking into this quite a bit